src/components/CamoComponent.vue

```diff
@@ -0,0 +1,211 @@
+<template>
+	<div class="camo">
+		<div
+			:key="camo.name"
+			:class="['camouflage']"
+			@click="handleToggleCompleted(camo)"
+			:content="requirementTooltip(camo)"
+			v-tippy="{ placement: 'bottom' }">
+			<div :class="['inner', this.isCompleted ? 'completed' : '']">
+				<img
+				:src="imageUrl(camo.name)"
+				:alt="camo.name"
+				onerror="javascript:this.src='/base-gradient.svg'" />
+				<IconComponent class="complete" name="check" fill="#10ac84" />
+				<IconComponent class="remove" name="times" fill="#ee5253" />
+				<span>
+					{{ camo.name }}
+				</span>
+			</div>
+		</div>
+	</div>
+</template>
+
+<script>
+import { useStore } from '@/stores/store'
+import { convertToKebabCase } from '@/utils/utils'
+import { mapActions, mapState } from 'pinia'
+
+export default {
+	data() {
+		return {
+			isCompleted: this.camo.isCompleted,
+		}
+	},
+	props: {
+		camo: {
+			type: Object,
+			required: true,
+		},
+	},
+	computed: {
+		...mapState(useStore, ['camoRequirements']),
+	},
+
+	methods: {
+		convertToKebabCase,
+		...mapActions(useStore, ['toggleCamouflageCompleted', 'toggleGoldCamouflageCompleted']),
+
+		imageUrl(camoName) {
+			if (camoName === 'Gold') {
+				return new URL('/gold-gradient.svg', import.meta.url)
+			} else if (camoName === 'Platinum') {
+				return new URL('/platinum-gradient.svg', import.meta.url)
+			} else if (camoName === 'Polyatomic') {
+				return new URL('/polyatomic-gradient.svg', import.meta.url)
+			}
+
+			return new URL(`../assets/camouflages/${convertToKebabCase(camoName)}.png`, import.meta.url)
+		},
```

Comment:
Can we change so this parameter is just named `camouflage`?
```suggestion
		imageUrl(camouflage) {
			if (camouflage === 'Gold') {
				return new URL('/gold-gradient.svg', import.meta.url)
			} else if (camouflage === 'Platinum') {
				return new URL('/platinum-gradient.svg', import.meta.url)
			} else if (camouflage === 'Polyatomic') {
				return new URL('/polyatomic-gradient.svg', import.meta.url)
			}

			return new URL(`../assets/camouflages/${convertToKebabCase(camouflage)}.png`, import.meta.url)
		},
```